sdk/appcenter-storage/src/main/java/com/microsoft/appcenter/storage/models/Document.java

```diff
@@ -31,40 +31,67 @@ public Document(T document, String partition, String id) {
         this.document = document;
     }
 
+    /**
+     * Create document from error.
+     * @param dError General exception.
```

Comment:
I don't know this general ;)
Just rename to exception with description as error when retrieving the document or something like that.